packages/core/src/transaction.ts

```diff
@@ -150,13 +150,13 @@ class Transaction {
               if (events != null) {
                 events.onError(error)
               }
-              return this._onError(error)
+              this._onError(error).catch(() => {})
```

Comment:
Why swallow the error? Are they handled in a different place or could this potentially make pinpointing a programming error in the driver (specifically in the error handler) quite hard?

Reply:
This error will never happen since `ConnectionHolder.releaseConnection` catch `Connection.release` errors. I've added logs (as warning) to failures on release connection in the `ConnectionHolder`. 

During the investigation, I realise a bug in the `Pool._release` and fixed. The Pool was not updating the resource count and resume pending requests if an error happens while destroying an resource. This behaviour can cause we run out of available connections in the pool if connections start to fail to be closed.